QgisModelBaker/utils/tools.py

```diff
@@ -0,0 +1,271 @@
+from QgisModelBaker.libs.modelbaker.generator.generator import Generator
+from QgisModelBaker.libs.modelbaker.iliwrapper import iliimporter
+from QgisModelBaker.libs.modelbaker.iliwrapper.globals import DbIliMode
+from QgisModelBaker.libs.modelbaker.iliwrapper.ili2dbconfig import (
+    ImportDataConfiguration,
+)
+from QgisModelBaker.libs.modelbaker.iliwrapper.ili2dbutils import JavaNotFoundError
+from QgisModelBaker.utils import gui_utils
+
+
+class QuickVisualizer(QObject):
+    """
+    This class provides the functionality for a quick import of dropped xtf files.
+    It generates a GeoPackage in the temporary directory withouth any constraints and imports the data without any validation.
+    The project layers are then generated with raw sql names.
+    """
+
+    def __init__(self, parent=None):
+        super().__init__(parent.iface.mainWindow())
+        self.parent = parent
+        self.message_bar = self.parent.iface.messageBar()
+
+    def push_message_bar(
+        self,
+        message,
+        running=False,
+        level=Qgis.MessageLevel.Info,
+        timeout=0,
+        show_log_button=False,
+    ):
+        """
+        Function to push a widget to the QGIS message bar containing info text and busy bar.
+        """
+        layout = QHBoxLayout()
+
+        if running:
+            busy_bar = QProgressBar()
+            busy_bar.setRange(0, 0)
+            busy_bar.setVisible(True)
+            layout.addWidget(busy_bar)
+
+        text_label = QLabel()
+        text_label.setText(message)
+        layout.addWidget(text_label)
+
+        if show_log_button:
+            button = QPushButton()
+            button.clicked.connect(lambda: self.parent.show_logs_folder())
+            button.setText(self.tr("Show log folder"))
+            layout.addWidget(button)
+
+        message_widget = QWidget()
+        message_widget.setLayout(layout)
+
+        self.message_bar.clearWidgets()
+        self.message_bar.pushWidget(message_widget, level, timeout)
+        self.log(f"Push to message bar: {message}", level)
+
+    def handle_dropped_files(self, dropped_files):
+
+        """
+        Main function receiving the files and trigger import to single GeoPackages.
+        After that it generates the layer via modelbaker generator.
+        """
+
+        data_files = []
+        model_files = []
+
+        # separate data files from ini files
+        for file in dropped_files:
+            if os.path.isfile(file):
+                if pathlib.Path(file).suffix[1:] in gui_utils.TransferExtensions:
+                    data_files.append(file)
+                else:
+                    model_files.append(file)
+
+        # one GeoPackage per file
+        status_map = {}
+        suc_files = set()
+        failed_files = set()
+
+        if len(data_files) == 0:
+            self.push_message_bar(
+                self.tr(
+                    "Nothing to import with Quick XTF Backer (use the Model Baker wizard instead)."
```

Comment:
*XTF Baker

Reply:
got rid of this name anyway. it's "Quick Visualizer"